src/lib/containers/table/table-top/DownloadOptions.tsx

```diff
@@ -24,80 +28,82 @@ type DownloadOptionsProps = {
 export const DOWNLOAD_FILES_MENU_TEXT = 'Add To Download Cart'
 const tooltipDownloadId = 'download'
 
-export const DownloadOptions: React.FunctionComponent<DownloadOptionsProps> = props => {
-  const { accessToken } = useSynapseContext()
-  const [showLoginModal, setShowLoginModal] = React.useState(false)
-  const [showExportMetadata, setShowExportMetadata] = React.useState(false)
-  const [showProgrammaticOptions, setShowProgrammaticOptions] = React.useState(
-    false,
-  )
-  const {
-    onDownloadFiles,
-    queryResultBundle,
-    queryBundleRequest,
-    darkTheme = true,
-  } = props
+export const DownloadOptions: React.FunctionComponent<DownloadOptionsProps> =
+  props => {
+    const { accessToken } = useSynapseContext()
+    const { entity } = useQueryContext()
+    const [showLoginModal, setShowLoginModal] = React.useState(false)
+    const [showExportMetadata, setShowExportMetadata] = React.useState(false)
+    const [showProgrammaticOptions, setShowProgrammaticOptions] =
+      React.useState(false)
+    const {
+      onDownloadFiles,
+      queryResultBundle,
+      queryBundleRequest,
+      darkTheme = true,
+    } = props
+
+    const isFileViewOrDataset =
+      entity &&
+      ((isEntityView(entity) && isFileView(entity)) || isDataset(entity))
```

Comment:
((isEntityView(entity) && hasFilesInView(entity)) || isDataset(entity))  from above is different from      ((isEntityView(entity) && isFileView(entity)) || isDataset(entity)) ? 
 Just confirming.  If we have more repetitions might be worth to have a unitlity fn instead of repeating the checks

Reply:
isFileView and hasFilesInView are slightly different; isFileView checks if the viewBitMask === 0x01 (has only files) where hasFilesInView checks if the bitmask includes 0x01 (has files and may have other things too)